src/librustc_typeck/check/mod.rs

```diff
@@ -2943,12 +2943,19 @@ fn check_expr_with_expectation_and_lvalue_pref<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
 
         if let Some((did, field_ty)) = private_candidate {
             let struct_path = fcx.tcx().item_path_str(did);
-            let msg = format!("field `{}` of struct `{}` is private", field.node, struct_path);
-            fcx.tcx().sess.span_err(expr.span, &msg);
             fcx.write_ty(expr.id, field_ty);
+            let msg = format!("field `{}` of struct `{}` is private", field.node, struct_path);
+            let mut err = fcx.tcx().sess.struct_span_err(expr.span, &msg);
+            // Also check if an accessible method exists, which is often what is meant.
+            if method::exists(fcx, field.span, field.node, expr_t, expr.id, false) {
+                err.note(&format!("a method called `{}` also exists, \
+                                   maybe a `()` to call it is missing?",
```

Comment:
nit: ``a method named `{}` `` or just ``a method `{}` ``

Also, I would prefer `perhaps you wish to call it` (instead of ``maybe a `()` to call it is missing?``) as in [this error message](https://github.com/rust-lang/rust/blob/8e414e0e3f27d1917d11ee80de827698beb53891/src/librustc_typeck/check/method/suggest.rs#L150).